File: csharp/CleanTests/Customer/CustomerService.cs
using LanguageExt;

namespace CleanTests.Customer;

public static class CustomerService
{
    public static Try<Store> Purchase(
        Store store,
        ProductType product,
        int quantity)
        => () => !store.HasEnoughInventory(product, quantity)
            ? throw new ArgumentException("Not enough inventory")
            : store.RemoveInventory(product, quantity);
}

File: php/src/Customer/CustomerService.php
<?php

namespace Com\CleanTests\Customer;

use http\Exception\InvalidArgumentException;

class CustomerService
{
    function purchase(Store       $store,
                      ProductType $productType,
                      int         $quantity): void
    {
        if (!$store->hasEnoughInventory($productType, $quantity))
            throw new InvalidArgumentException("Not enough inventory");
        else $store->removeInventory($productType, $quantity);
    }
}